Handle MAV_CMD_DO_SET_MISSION_CURRENT

peterbarker opened the pull request:
We have a normal message (`MISSION_SET_CURRENT`) which we use to do this, relying on the `CURRENT_CURRENT` message to provide confirmation of the change.

There does exist this MAV_CMD_DO_SET_MISSION_CURRENT message which will provide an acknowledgement, however neither QGC nor the `PX4 Firmware` project appear to have support for it in their master branch.

It is a state change on the aircraft, and those are supposed to be accomplished with commands rather than messages....

tridge commented on libraries/GCS_MAVLink/GCS_Common.cpp:
```diff
@@ -3767,6 +3767,24 @@ MAV_RESULT GCS_MAVLINK::handle_command_preflight_can(const mavlink_command_long_
+    }
+
+    const uint32_t seq = (uint32_t)packet.param1;
+    if (!mission->set_current_cmd(seq)) {
```
does this handle seq==0 for resetting mission?

davidbuzz commented:
might be worth a comment in the code that this is an alternative to a more common packet.?

peterbarker commented:
I've added some comments to clarify old and new methods.

I've added the extra handling for Plane for when the waypoint is changed via mavlink.

I've got concerns about scripting's access to the mission library; it may not reset Plane's state correctly if it is only dealing with the Mission library's set-waypoint command.
